cmd/kube-apiserver/app/server.go

```diff
@@ -461,7 +454,7 @@ func BuildGenericConfig(
 	kubeVersion := version.Get()
 	genericConfig.Version = &kubeVersion
 
-	storageFactory, lastErr := BuildStorageFactory(s, genericConfig.MergedResourceConfig)
+	storageFactory, lastErr = storageFactoryConfig.Complete().New()
```

Comment:
Ok, your pull is making some interesting things obvious.  `StorageEncodingOverrides` seems to be crux of what we're getting from the `StorageSerializationOptions`.  Also, the New method is returning a standard type `*serverstorage.DefaultStorageFactory`.  

As a stepwise refinement from here to there, could your `storageFactorConfig.Complete` take the `StorageSerialization` and the `EtcdOptions` to fill in the required information?  We'd still be slightly cyclical, but I think it would start extracting what information we're combining.